Mathlib/Algebra/Order/Star/Basic.lean

```diff
@@ -166,6 +166,10 @@ theorem star_mul_self_nonneg (r : R) : 0 ≤ star r * r :=
 theorem mul_star_self_nonneg (r : R) : 0 ≤ r * star r := by
   simpa only [star_star] using star_mul_self_nonneg (star r)
 
+@[aesop safe apply (rule_sets := [CStarAlgebra])]
+theorem IsSelfAdjoint.mul_self_nonneg {a : R} (ha : IsSelfAdjoint a) : 0 ≤ a * a := by
+  simpa [ha.star_eq] using star_mul_self_nonneg a
```

Comment:
Thanks! Can you add a `IsSelfAdjoint.sq_nonneg : 0 ≤ a ^ 2` version too? In that case you'll need to assume the `R` is unital (you can put the result in the `Semiring` section below.

Reply:
Will do this before finishing the PR. Thanks!